src/Components/Modal/ModalLayout.js

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable */
+import React, { Component } from "react";
+import { withRouter } from "react-router";
+import styled from "styled-components";
+import ModalArrows from "./ModalArrows";
+import { Store } from "../Lecture/LectureForm";
+
+class ModalLayout extends Component {
+  state = {
+    id: 0,
+    imgLength: 0,
+    intervalId: null,
+  };
+  componentDidMount = () => {
+    const { images } = this.props.data;
+    if (this.props.type === "openSoon") {
+      const intervalId = setInterval(() => {
+        this.handleNext();
+      }, 2000);
+      this.setState({
+        intervalId,
+      });
+    }
+    this.setState({
+      imgLength: images.length - 1,
+    });
+  };
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    if (
+      this.state.id === this.state.imgLength &&
+      this.state.intervalId !== null
+    ) {
+      clearInterval(this.state.intervalId);
+      this.setState({ intervalId: null });
+    }
+    if (
+      this.state.intervalId === null &&
+      this.state.id < this.state.imgLength
+    ) {
+      const intervalId = setInterval(() => {
+        this.handleNext();
+      }, 2000);
+      this.setState({
+        intervalId: intervalId,
+      });
+    }
+  }
+  handleAddClassName = id => {
+    if (this.state.id >= id) {
+      return "on";
+    }
+  };
```

Comment:
1. return 값이 없는 함수는 무엇을 반환할까요?
2. `conosole.log(undefined+'myClassName')`의 결과값을 확인해주세요.
3. `console.log(null+'anotherClassName')`의 값도 확인해주세요.

위 질문을 바탕으로 아래 질문에 답해주세요.
현재는 className을 하나로만 하고 있지만
유동적으로 className을 줄 때 발생할 문제가 있습니다.
그 문제와 함께 아래와 같은 코드로 바꾸어야 하는 이유를 코멘트로 달아주세요.
단순히 간단한 코드로 바꾸기 위함 아닙니다!

```suggestion
  handleAddClassName = id => this.state.id>=id ? 'on' : '' };
```

Reply:
1. return 값이 없는 함수는 무엇을 반환할까요?
conosole.log(undefined+'myClassName')의 결과값을 확인해주세요.
console.log(null+'anotherClassName')의 값도 확인해주세요.